File: treeClass.py
class node(object):
    def __init__(self, value, parent=None, children=[]):
        self.value = value
        self.parent = None
        self.children = []

    def getData(node):
        return node.value

    def __str__(self, level=-1):
        printString = '-' * level + str(self.value) + '\n'
        sortedChildren = sorted(self.children, key=node.getData)
        for child in sortedChildren:
            printString += child.__str__(level + 1)
        return printString

File: c#_code_challenge/code_challenge_rewrite/treeClass.cs
using System;
using System.Collections.Generic;

namespace code_challenge_rewrite
{
    public class Node
    {
        public string name;
        public Node parent;
        public List<Node> children;

        public Node(string value)
        {
            name = value;
            parent = null;
            children = new List<Node>();
        }

        public string ToString(int level = -1)
        {
            string printString = "";
            for (int i = 0; i < level; i++)
            {
                printString = $"{new String('-', level)}";
            }
            printString += $"{name}\n";

            children.Sort((a, b) => (a.name.CompareTo(b.name)));

            foreach (var child in children)
            {
                printString += child.ToString(level + 1);
            }
            return printString;
        }
    }
}